src/main/java/beam/analysis/plots/DeadHeadingAnalysis.java

```diff
@@ -119,10 +119,10 @@ public void resetStats() {
     private void processDeadHeading(Event event) {
```

Comment:
Rather than casting internally, the parameter type should be changed so that it can be caught earlier at compile time.

Reply:
Agree, but I'm not an author of that handler, maybe someone wanted to have an ability to process any type of events there. But for now will make it parameterzed explicitly to that type.

Reply:
But you are casting, so if it is NOT that type, then it will blow up at runtime. Since that is known we should make it blow up at compile time.